src/meshes/p4est_mesh.jl

```diff
@@ -1597,6 +1871,128 @@ function calc_tree_node_coordinates!(node_coordinates::AbstractArray{<:Any, 4},
     return file_idx
 end
 
+# TODO: 3D version
+# Version for quadratic 2D elements, i.e., second-order quads.
+function calc_tree_node_coordinates!(node_coordinates::AbstractArray{<:Any, 4},
+                                     element_lines, nodes, vertices, RealT,
+                                     linear_quads, mesh_nodes)
+    nnodes = length(nodes)
+
+    # Create a work set of Gamma curves to create the node coordinates
+    CurvedSurfaceT = CurvedSurface{RealT}
+    surface_curves = Array{CurvedSurfaceT}(undef, 4)
+
+    # Create other work arrays to perform the mesh construction
+    element_node_ids = Array{Int}(undef, 4)
+    quad_vertices = Array{RealT}(undef, (4, 2))
+    curve_values = Array{RealT}(undef, (nnodes, 2))
+
+    # Create the barycentric weights used for the surface interpolations
+    bary_weights_ = barycentric_weights(nodes)
+    bary_weights = SVector{nnodes}(bary_weights_)
+
+    element_set_order = 1
+    tree = 0
+    for line_idx in 1:length(element_lines)
+        line = element_lines[line_idx]
+
+        # Check if a new element type/element set is defined
+        if startswith(line, "*ELEMENT")
+            # Retrieve element type
+            current_element_type = match(r"\*ELEMENT, type=([^,]+)", line).captures[1]
+
+            # Check if these are linear elements
+            if occursin(linear_quads, current_element_type)
+                element_set_order = 1
+            else
+                element_set_order = 2
+            end
+        else # Element data
+            tree += 1
+
+            # Pull the vertex node IDs
+            line_split = split(line, r",\s+")
+            element_nodes = parse.(Int, line_split)
+
+            if element_set_order == 1
+                element_node_ids[1] = element_nodes[2]
+                element_node_ids[2] = element_nodes[3]
+                element_node_ids[3] = element_nodes[4]
+                element_node_ids[4] = element_nodes[5]
+
+                # Create the node coordinates on this particular element
+                # Pull the (x,y) values of the four vertices of the current tree out of the global vertices array
+                for i in 1:4
+                    quad_vertices[i, :] .= vertices[1:2, element_node_ids[i]] # 2D => 1:2
+                end
+                calc_node_coordinates!(node_coordinates, tree, nodes, quad_vertices)
+            else # element_set_order == 2
+                for edge in 1:4
+                    node1 = element_nodes[edge + 1] # "Left" node
+                    node2 = element_nodes[edge + 5] # "Middle" node
+                    node3 = edge == 4 ? element_nodes[2] : element_nodes[edge + 2] # "Right" node
+
+                    node1_coords = mesh_nodes[node1]
+                    node2_coords = mesh_nodes[node2]
+                    node3_coords = mesh_nodes[node3]
+
+                    # The nodes for an Abaqus element are labeled following a closed path 
+                    # around the element:
```

Comment:
Hmm, I imagine that figuring out this variable flipping will be significantly more cumbersome in 3D and for higher node orders.

Reply:
Yeah, that is also why the 3D version is not shipped in this PR. I think one needs in that case to flip the faces also in some way which I have not yet figured out.

Reply:
I totally understand keeping the 3D for a future PR. This was more me thinking out loud that the 3D flips and logic would be quite ugly.